dockers/docker-dhcp-server/cli/config/plugins/dhcp_server.py

```diff
@@ -1,13 +1,106 @@
 import click
 import utilities_common.cli as clicommon
 
+import ipaddress
+
+
+SUPPORT_TYPE = ["binary", "boolean", "ipv4-address", "string", "uint8", "uint16", "uint32"]
+
+
+def validate_str_type(type, value):
+    """
+    To validate whether type is consistent with string value
+    Args:
+        type: string, value type
+        value: checked value
+    Returns:
+        True, type consistent with value
+        False, type not consistent with value
+    """
+    if not isinstance(value, str):
+        return False
+    if type not in SUPPORT_TYPE:
+        return False
+    if type == "string":
+        return True
+    if type == "binary":
+        if len(value) == 0 or len(value) % 2 != 0:
+            return False
+        return all(c in set(string.hexdigits) for c in value)
+    if type == "boolean":
+        return value in ["true", "false"]
+    if type == "ipv4-address":
+        try:
+            if len(value.split(".")) != 4:
+                return False
+            return ipaddress.ip_address(value).version == 4
+        except ValueError:
+            return False
+    if type.startswith("uint"):
+        if not value.isdigit():
+            return False
+        length = int("".join([c for c in type if c.isdigit()]))
+        return 0 <= int(value) <= int(pow(2, length)) - 1
+    return False
+
 
 @click.group(cls=clicommon.AbbreviationGroup, name="dhcp_server")
 def dhcp_server():
     """config DHCP Server information"""
+    ctx = click.get_current_context()
+    dbconn = db.db
+    if dbconn.get("CONFIG_DB", "FEATURE|dhcp_server", "state") != "enabled":
+        ctx.fail("Feature dhcp_server is not enabled")
+
+
+@dhcp_server.group(cls=clicommon.AliasedGroup, name="ipv4")
+def dhcp_server_ipv4():
+    """Show ipv4 related dhcp_server info"""
     pass
 
 
+@dhcp_server_ipv4.command(name="add")
+@click.argument("dhcp_interface", required=True)
+@click.option("--mode", required=True)
+@click.option("--lease_time", required=False, default="900")
+@click.option("--dup_gw_nm", required=False, default=False, is_flag=True)
+@click.option("--gateway", required=False)
+@click.option("--netmask", required=False)
+@clicommon.pass_db
+def dhcp_server_ipv4_add(db, mode, lease_time, dup_gw_nm, gateway, netmask, dhcp_interface):
+    ctx = click.get_current_context()
+    if mode != "PORT":
+        ctx.fail("Only mode PORT is supported")
```

Comment:
Could you add unit test to cover `ctx.fail` scenarios in line 53, 73, 75, 88, 90?